pages/extensions/amp_example_preview/util/preview.py

```diff
@@ -0,0 +1,98 @@
+import re
+import json
+
+PREVIEW_START_BEGINNING = '<!-- preview\n'
+PREVIEW_END_TAG = r'<!-- /preview -->'
+PREVIEW_PATTERN = re.compile(PREVIEW_START_BEGINNING + r'(.*?)\n-->\n(.*?)' + PREVIEW_END_TAG, re.DOTALL)
+
+
+class ExamplePreview(object):
+
+    def get_start_tag(self):
+        jsonData = json.dumps(self.__dict__)
+        return PREVIEW_START_BEGINNING + escape_html(jsonData) + '\n-->\n'
+
+    def get_end_tag(self):
+        return PREVIEW_END_TAG
+
+    @staticmethod
+    def from_json(jsonData):
+        """
+        :type jsonData: str
+        """
+        result = ExamplePreview()
+        result.__dict__ = json.loads(unescape_html(jsonData))
+        return result
+
+    @staticmethod
+    def for_attributes(index, mode, playground, source):
```

Comment:
Encapsulate all options in an object to simplify your code. With the current approach, adding a new flag requires changing code in multiple locations.

Reply:
Done